### PR TITLE
User account creation email change

### PR DESCRIPTION
**Pull request by akssinhaamit**
# Motivation and Context

# What has changed

# How to test?

# Links

# Screenshots (if appropriate):

### CODE REVIEW

File: frontstage/views/account/account.py [hunk 1]
```diff
@@ -49,6 +68,7 @@ def change_account_details(session):
                                                       attributes_changed,
                                                       respondent_details,
                                                       update_required_flag)
+        email_update_required_flag = form['email_address'].data != respondent_details['emailAddress']
```
Comment (insacuri) [inline frontstage/views/account/account.py:71]: Can you change all instances of the following variables in this file:
- `email_update_required_flag` to `is_email_update_required`
- `update_required_flag` to `is_update_required`

Its minor change but I think language is important.

**Review by insacuri — CHANGES_REQUESTED** [review]
A few changes but it's looking good!

File: frontstage/controllers/party_controller.py [hunk 2]
```diff
@@ -177,6 +177,19 @@ def resend_verification_email_expired_token(token):
...
 
+def resend_account_email_change_expired_token(token):
+    logger.info('Re-sending account email change verification email', token=token)
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/resend-account-email-change-expired-token/{token}'
+    response = requests.post(url, auth=app.config['BASIC_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        logger.error('Re-sending of verification email for expired token failed')
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/controllers/party_controller.py]: 👍

File: frontstage/templates/account/account-change-email-address.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        {{
+            onsButton({
+            "text": "Send",
+            "id": "btn-option-send-email",
+            "classes": 'btn-group__btn',
+            "submitType": "timer"
+            })
+        }}
+        <a href="{{ url_for('account_bp.change_account_details') }}"
+      role="button" class="btn btn--secondary" id="btn-upload-ci-cancel"><span class="btn__inner">Cancel</span></a>
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/templates/account/account-change-email-address.html]: 👍

File: frontstage/views/account/account.py [hunk 4]
```diff
@@ -35,6 +35,25 @@ def update_account(session):
...
 
+@account_bp.route('/change-account-email-address', methods=['POST'])
+@jwt_authorization(request)
+def change_email_address(session):
+    form = ConfirmEmailChangeForm(request.values)
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
+    respondent_details['email_address'] = respondent_details['emailAddress']
+    respondent_details['new_email_address'] = form['email_address'].data
+    respondent_details['respondent_initiated'] = True
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/views/account/account.py]: 👍

File: frontstage/views/account/account.py [hunk 5]
```diff
@@ -35,6 +35,25 @@ def update_account(session):
...
+    logger.info('Attempting to update email address changes on the account')
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/views/account/account.py]: 👍

File: frontstage/views/account/account.py (hunk 1 above)
Comment (akssinhaamit) [reply frontstage/views/account/account.py]: 👍

File: frontstage/views/account/account.py [hunk 6]
```diff
@@ -43,14 +43,14 @@ def change_email_address(session):
...
     respondent_details['new_email_address'] = form['email_address'].data
-    respondent_details['respondent_initiated'] = True
-    logger.info('Attempting to update email address changes on the account')
+    respondent_details['change_requested_by_respondent'] = True
+    logger.info('Attempting to update email address changes on the account', party_id=party_id)
     try:
         party_controller.update_account(respondent_details)
     except ApiError as exc:
-        logger.error('Failed to updated email on account', status=exc.status_code)
+        logger.error('Failed to updated email on account', status=exc.status_cod, party_id=party_id)
```
Comment (insacuri) [inline frontstage/views/account/account.py:51]: `status_cod` should be `status_code`

Comment (akssinhaamit) [reply frontstage/views/account/account.py:51]: :-) True

**Review by insacuri — APPROVED** [review]

**Pull request merged**